File: main1.py
integer = int(input('Введите целое число: '))
fl = float(input('Введите дробное число: '))
string = input('Введите любое слово: ')

print(f'{integer} - {type(integer).__name__}')
print(f'{fl} - {type(fl).__name__}')
print(f'{string} - {type(string).__name__}')

File: main1.cpp
#include <iostream>
#include <string>
#include <typeinfo>

int main() {
  int integer;
  double decimal;
  std::string str;

  std::cout << "\n";
  std::cout << "Введите целое число: ";
  std::cin >> integer;

  std::cout << "Введите дробное число: ";
  std::cin >> decimal;

  std::cin.ignore();

  std::cout << "Введите любое слово: ";
  std::getline(std::cin, str);

  std::cout << "\n";
  std::cout << integer << " - " << typeid(integer).name() << "\n";
  std::cout << decimal << " - " << typeid(decimal).name() << "\n";
  std::cout << str << " - " << typeid(str).name() << "\n";

  return 0;
}